provision/provision.py

```diff
@@ -0,0 +1,157 @@
+es = boto3.client('es')
+
+
+def getpath(filename):
+    return os.path.abspath(os.path.join(os.path.dirname(__file__), filename))
+
+
+def get_legacy_client(endpoint):
+    sys.path.insert(0, getpath('..'))
+    os.environ['ES_HOST'] = endpoint[7:]  # remove http://
+    from app import client
+    return client
+
+
+def get_endpoint(es_domain):
+    """
+    Finds the endpoint (that requests should be made to) for a given
+    ElasticSearch domain.
+
+    :rtype: string
+    """
+    backoff = 0
+    status = {}
+    while not status.get('DomainStatus', {}).get('Endpoint', None):
+        status = es.describe_elasticsearch_domain(DomainName=es_domain)
+        # \sum_{n=0}^{9} 2^n is a long time and if there hasn't been a response
+        # by then, one probably isn't coming anytime soon.
+        if backoff > 10:
+            raise RuntimeError("It's been a long time, and the ElasticSearch "
+                               "domain still looks to be unavailable. It's "
+                               "been long enough that I'm considering this "
+                               "a timeout.")
+        if not status['DomainStatus'].get('Endpoint', None):
+            time.sleep(2 ** backoff)
+            backoff += 1
+
+    # boto3 returns the endpoint without the URL protocol, so we need
+    # to add it if it doesn't exist, otherwise requests will choke
+    return 'http://' + status['DomainStatus']['Endpoint']
+
+
+def populate_domain(endpoint):
+    """
+    Given an ElasticSearch endpoint, populates the domain with a
+    'faithful' replica of dss-azul-commons for testing purposes.
+    """
+    # boto3 can't sign requests by default, so we fall back to boto2
+    client = get_legacy_client(endpoint)
+
+    # Set up fb_index (data_objects)
+    with open(getpath('index-mapping.json'), 'r') as data:
+        payload = json.load(data)['fb_index']
+    client.make_request(method='PUT', path='/fb_index', data=json.dumps(payload))
+
+    # Set up db_index (data bundles)
+    with open(getpath('index-mapping.json'), 'r') as data:
+        payload = json.load(data)['db_index']
+    client.make_request(method='PUT', path='/db_index', data=json.dumps(payload))
+
+    # Populate both indexes
+    with open(getpath('test-data.json'), 'r') as data:
+        client.make_request(method='POST', path='/_bulk', data=data.read())
+
+
+def raze_domain(endpoint):
+    """
+    Given an ElasticSearch endpoint, drops fb_index and db_index so that
+    it can be populated anew.
+    """
+    # boto3 can't sign requests by default, so we fall back to boto2
+    client = get_legacy_client(endpoint)
+
+    # Drop fb_index
+    client.make_request(method='DELETE', path='/fb_index')
+
+    # Drop db_index
+    client.make_request(method='DELETE', path='/db_index')
+
+
+def setup(es_domain=None):
+    """
+    Sets up an ElasticSearch domain.
+    """
+    if not es_domain:
+        es_domain = 'dos-azul-test-' + str(uuid.uuid1()).split('-')[0]
```

Comment:
Here, you're actually creating the ES domain if it wasn't provided. Very cool feature and beats clicking on AWS console any day.